src/popup/lib/utils.ts

```diff
@@ -62,3 +64,59 @@ export function formatNumber(
     } else return value.toExponential()
   }
 }
+
+export function useLocalStorage<T>(
+  key: string,
+  {
+    defaultValue,
+    maxAge = Infinity
+  }: Partial<{ defaultValue: T; maxAge: number }> = {}
+) {
+  const hasLocalStorage = typeof localStorage !== 'undefined'
+  type Stored = { value: T; ts: number }
+  const isWellFormed = (obj: any): obj is Stored => {
+    if (typeof obj !== 'object' || obj == null) return false
+    if (!obj.ts || !Number.isSafeInteger(obj.ts)) return false
+    return typeof obj.value !== 'undefined'
+  }
```

Comment:
We will probably not have a lot of re-renders, but I think we can wrap this in a `useCallback`.